src/neo4j/addressing.py

```diff
@@ -79,22 +79,83 @@ def ipv6_cls(self):
 
 
 class Address(tuple, metaclass=_AddressMeta):
+    """Base class to represent server addresses within the driver.
+
+    A tuple of two (IPv4) or four (IPv6) elements, representing the address
+    parts. See also python's :mod:`socket` module for more information.
```

Comment:
Is it lengthy to explain what the full signature is? If not, I would consider adding it. I am confused by `Address(("::1", 7687, 0, 0))` as it's not immediate what the third and fourth arguments do.

Reply:
It is rather involved depending on how deep you want to dig into the rabbit hole. The full IPv6 signature is `(host, port, flowinfo, scope_id)` (as of Python's `socket` module documentation wich I linked here). There are two things to consider here:
 1) flow info and scope id are well beyond the scope of the driver's API docs, so I def. wouldn't go deeper than that.
 2) the driver uses this 4-tuple and, for the most parts, passes it as-is into Python built-in APIs. So by keeping this generic, an admittedly unlikely change in those APIs will transparently propagate through the driver's `Address` classes.

If you think the sole signature of the tuples (in the form I wrote above) might be helpful to uses, I'm happy to include that in the docs.

Reply:
I hoped the signature alone to be more descriptive, but at a glance I have no idea of what `flowinfo` and `scope_id` would be, so I don't think it would be of use to 80% of other people. We can drop this.